app/bin/configuration.dart

```diff
@@ -50,3 +50,25 @@ class Configuration {
       : projectId = 'dartlang-pub-dev',
         packageBucketName = 'dartlang-pub-dev--pub-packages';
 }
+
+/// Configuration from the environment variables.
+class EnvConfig {
+  final String gcloudKey;
+  final String gcloudProject;
+
+  EnvConfig._({this.gcloudKey, this.gcloudProject});
+
+  factory EnvConfig._detect() {
+    return new EnvConfig._(
+      gcloudKey: Platform.environment['GCLOUD_KEY'],
+      gcloudProject: Platform.environment['GCLOUD_PROJECT'],
+    );
+  }
+
+  bool get hasGcloudKey => gcloudKey != null;
```

Comment:
This getter is unnecessary, since we always require `GCLOUD_PROJECT` to be there.